src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,156 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+import java.util.Objects;
+
 public class ArrayList<T> implements List<T> {
+    public static final int DEFAULT_ARRAY_CAPACITY = 10;
+    private static final double INCREASING_STEP = 1.5;
+    private Object[] values;
+    private int size;
+
+    public ArrayList() {
+        values = new Object[DEFAULT_ARRAY_CAPACITY];
+        size = 0;
+    }
+
     @Override
     public void add(T value) {
-
+        if (size == values.length) {
```

Comment:
It's better to change grow() method and include if (size == values.length) checking in this method to avoid duplicating of code. 
See grow() method comment